components/espsense/espsense.h

```diff
@@ -10,34 +13,49 @@
 #include "ESPAsyncUDP.h"
 #endif
 
+namespace esphome {
+namespace espsense {
+
 #define RES_SIZE 400
 #define REQ_SIZE 70
 #define MAX_PLUG_COUNT 10  // Somewhat arbitrary as of now
 
 class ESPSensePlug {
-public:
+ public:
   std::string name;
   std::string mac;
   bool encrypt = true;
   float voltage = 120.0;
-  Sensor *power_sid = NULL;
-  Sensor *voltage_sid = NULL;
-  Sensor *current_sid = NULL;
+  sensor::Sensor *power_sid = NULL;
+  sensor::Sensor *voltage_sid = NULL;
+  sensor::Sensor *current_sid = NULL;
   
   std::string base_json = "{\"emeter\": {\"get_realtime\":{ "
                               "\"current\": %.02f, \"voltage\": %.02f, \"power\": %.02f, \"total\": 0, \"err_code\": 0}}, "
                            "\"system\": {\"get_sysinfo\": "
                               "{\"err_code\": 0, \"hw_ver\": 1.0, \"type\": \"IOT.SMARTPLUGSWITCH\", \"model\": \"HS110(US)\", "
                            "\"mac\": \"%s\", \"deviceId\": \"%s\", \"alias\": \"%s\", \"relay_state\": 1, \"updating\": 0 }}}";
-  
-  
-  ESPSensePlug(Sensor *sid, std::string config_mac, std::string config_name, float config_voltage) {
-    power_sid = sid;
-    mac = config_mac;
-    name = config_name;
-    voltage = config_voltage;
+
+  ESPSensePlug() {}
+
+  void set_name(std::string name) { 
+    this->name = name;
+
+    // Generate a fake MAC address from the name to prevent issues when there are multiple "plugs" with the same MAC address
```

Comment:
I'm on the fence about always generating a MAC from the name, because it's a departure from the current approach of using the hardware MAC by default unless otherwise specified. See other comment below on the `addPlug` function for an alternate approach.